packages/react-devtools-core/src/deviceStorage.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import {
+  type ConsolePatchSettings,
+  writeConsolePatchSettingsToWindow,
+} from 'react-devtools-shared/src/backend/console';
+import {
+  LOCAL_STORAGE_SHOULD_APPEND_COMPONENT_STACK_KEY,
+  LOCAL_STORAGE_SHOULD_BREAK_ON_CONSOLE_ERRORS,
+  LOCAL_STORAGE_SHOW_INLINE_WARNINGS_AND_ERRORS_KEY,
+  LOCAL_STORAGE_HIDE_CONSOLE_LOGS_IN_STRICT_MODE,
+  LOCAL_STORAGE_BROWSER_THEME,
+} from 'react-devtools-shared/src/constants';
+import type {BrowserTheme} from 'react-devtools-shared/src/devtools/views/DevTools';
+
+export type DeviceStorageMethods = {|
+  setValueOnDevice: (key: string, value: string) => void,
```

Comment:
Nit: Can we give this a similar API to the local storage functions? 
```js
    export type DeviceStorage = {|
       getItem: (key: string) => ?string,
       setItem: (key: string, value: any) => void,
       removeItem: (key: string) => void (I think?)
    |}
```

Reply:
Just chatted — this doesn't apply to the new "entire blob at once" version